### PR TITLE
Fix Tokenizer Inference

### PR DESCRIPTION
**Pull request by pbontrager**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [x] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Qwen base tokenizer __call__ method didn't pass the inference flag into tokenize_messages. This affected all Qwen models. Issue reported on [discord](https://discord.com/channels/1216353675241590815/1216353675744641096/1374658007878598699).

#### Changelog
Update __call__ to not add_end_tokens when inference==True. Update naming in tokenizer to use add_end_tokens instead of add_eos.

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [x] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [x] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [x] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [x] manually run any new or modified recipes with sufficient proof of correctness
- [x] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

Ran user provided config for Qwen2, Qwen2.5, and Qwen3:
```
output_dir: ./ # Not needed
base_model_dir: /tmp/Qwen2.5-0.5B-Instruct

# Model arguments
model:
  _component_: torchtune.models.qwen2_5.qwen2_5_0_5b

checkpointer:
  _component_: torchtune.training.FullModelHFCheckpointer
  checkpoint_dir: ${base_model_dir}
  checkpoint_files: [model.safetensors]
  recipe_checkpoint: null
  output_dir: ${output_dir}
  model_type: QWEN2

device: cuda
dtype: bf16

seed: 1234

# Tokenizer arguments
tokenizer:
  _component_: torchtune.models.qwen2_5.qwen2_5_tokenizer
  path: ${base_model_dir}/vocab.json
  merges_file: ${base_model_dir}/merges.txt
  max_seq_len: null

# Generation arguments; defaults taken from gpt-fast
prompt:
  system: null
  user: "Tell me a joke."
max_new_tokens: 300
temperature: 0.6 # 0.8 and 0.6 are popular values to try
top_k: 300

enable_kv_cache: True
quantizer: null
```

### CODE REVIEW

**Review by joecummings — APPROVED** [review]
Can you paste teh output from running a config too?

**Comment by pbontrager** [comment]
@joecummings  you want to hear the jokes?

Qwen 2.5
```
Tell me a joke.
assistant
Why couldn't the bicycle stand up by itself?

Because it was two-tired!
```

Other Qwens had similar cheesy jokes but I lost the history.
